extensions/emoji/src/emoji.tsx

```diff
@@ -0,0 +1,110 @@
+import {
+  ActionPanel,
+  CopyToClipboardAction,
+  PasteAction,
+  List,
+  getLocalStorageItem,
+  setLocalStorageItem,
+} from "@raycast/api";
+
+import { useState, useEffect, useRef, useCallback } from "react";
+import type { ReactElement, SetStateAction, Dispatch } from "react";
+import { createEmojiList } from "generate-emoji-list";
+
+type Category = { category: string; emojis: Emoji[] };
+type Emoji = {
+  emoji: string;
+  description: string;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const useStateFromLocalStorage = <T, _ = void>(key: string, initialValue: T): [T, Dispatch<SetStateAction<T>>] => {
+  const [state, setState] = useState<T>(initialValue);
+  const didUnmount = useRef<boolean>(false);
+
+  useEffect(() => {
+    (async () => {
+      const cache = await getLocalStorageItem(key);
+
+      if (typeof cache === "string") {
+        if (!didUnmount.current) {
+          setState(JSON.parse(cache));
+        }
+      }
+    })();
+
+    return () => {
+      didUnmount.current = true;
```

Comment:
Using `didUnmount` looks really complex, what are you using this for? To implement some kind of caching?

Reply:
I just want to make sure the async code doesn't result in state being set on an unmounted component.